src/core/lv_obj_class_private.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file lv_obj_class_private.h
+ *
+ */
+
+#ifndef LV_OBJ_CLASS_PRIVATE_H
+#define LV_OBJ_CLASS_PRIVATE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_obj_class.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**
+ * Describe the common methods of every object.
+ * Similar to a C++ class.
+ */
+struct lv_obj_class_t {
```

Comment:
by making this class private you are removing the ability for people to make custom widgets.

Reply:
New widgets can be created in C. I'm not sure it's something that we should support in the bindings.

Reply:
Let's discuss creating new MP widgets separately too.